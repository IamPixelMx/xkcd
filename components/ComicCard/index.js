import ComicCard from './ComicCard';

export { ComicCard };
