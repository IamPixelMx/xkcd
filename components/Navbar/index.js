import Navbar from './Navbar';
import NavbarInput from './NavbarInput';
import NavbarItem from './NavbarItem';

export { Navbar, NavbarInput, NavbarItem };
