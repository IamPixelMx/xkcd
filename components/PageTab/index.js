import PageTab from './PageTab';

export { PageTab };
