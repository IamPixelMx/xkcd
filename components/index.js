import { Footer } from './Footer';
import { Navbar } from './Navbar';
import { PageTab } from './PageTab';

export { Footer, Navbar, PageTab };
