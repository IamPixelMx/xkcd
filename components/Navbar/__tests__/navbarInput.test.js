import React from 'react';
import * as ReactReduxHooks from 'react-redux';
import configureStore from 'redux-mock-store';
import { fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import { renderWithProvider, renderWithProviderSnapshot } from 'jest-utils';
import { NavbarInput } from 'components';

const dispatch = jest.fn();
jest.spyOn(ReactReduxHooks, 'useDispatch').mockImplementation(() => dispatch);

const mockStore = configureStore([]);

const navbarInputElement = <NavbarInput />;

describe('NavbarInput', () => {
  it('Renders NavbarInput ', () => {
    const navbarInput = renderWithProviderSnapshot(navbarInputElement, {
      mockStore,
    }).toJSON();
    expect(navbarInput).toMatchSnapshot();
  });

  it('Add and remove "is-hidden" class to help text', () => {
    const { getByRole, getByTestId } = renderWithProvider(navbarInputElement, {
      mockStore,
    });
    const input = getByRole('searchbox');
    const help = getByTestId('help');
    input.focus();
    expect(help).not.toHaveClass('is-hidden');
    input.blur();
    expect(help).toHaveClass('is-hidden');
  });

  it('change value input and dispatch value on "Enter"', async () => {
    const { getByRole } = renderWithProvider(navbarInputElement, {
      mockStore,
    });
    const input = getByRole('searchbox');
    fireEvent.change(input, { target: { value: '23' } });
    expect(input.value).toBe('23');
    await act(async () => {
      fireEvent.keyPress(input, { key: 'Enter', code: 13, charCode: 13 });
    });
    expect(dispatch).toBeCalledTimes(1);
  });
});
