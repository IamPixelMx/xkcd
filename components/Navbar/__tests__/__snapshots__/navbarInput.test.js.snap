// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarInput Renders NavbarInput  1`] = `
<div
  className="column"
>
  <div
    className="navbar-item field"
  >
    <div
      className="control has-icons-left"
    >
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder="Buscar cómic..."
        type="search"
        value=""
      />
      <span
        className="icon is-small is-left"
      >
        🔍
      </span>
      <p
        className="help has-text-grey has-text-weight-medium is-hidden"
        data-testid="help"
      >
        <small>
          Ingresa número/id de cómic
        </small>
      </p>
    </div>
  </div>
</div>
`;
